src/crypto/ecdsa.cpp

```diff
@@ -50,4 +51,24 @@ namespace crypto
     return ecdsa_sig_from_r_s(
       signature.data(), half_size, signature.data() + half_size, half_size);
   }
+
+  std::vector<uint8_t> ecdsa_sig_der_to_p1363(
+    const std::vector<uint8_t>& signature, CurveID curveId)
+  {
+    auto sig_ptr = signature.data();
+    OpenSSL::Unique_ECDSA_SIG ecdsa_sig(
+      d2i_ECDSA_SIG(NULL, &sig_ptr, signature.size()));
+    // r and s are managed by Unique_ECDSA_SIG object, so we shouldn't use
+    // Unique_BIGNUM for them
+    const BIGNUM* r = ECDSA_SIG_get0_r(ecdsa_sig);
+    const BIGNUM* s = ECDSA_SIG_get0_s(ecdsa_sig);
+    int nid = PublicKey_OpenSSL::get_openssl_group_id(curveId);
+    OpenSSL::Unique_EC_GROUP ec_group(nid);
+    int group_order_bits = EC_GROUP_order_bits(ec_group);
```

Comment:
A check that group_order_bits is positive would be good, just out of caution.

Reply:
Added it here [1e46300](https://github.com/microsoft/CCF/pull/4829/commits/1e46300951dc7f4a676836b84fb44241e1fca37d)